docker/tests/test_zap_hooks.py

```diff
@@ -0,0 +1,83 @@
+import unittest
+from datetime import datetime
+from contextlib import contextmanager
+import zap_common
+
+
+class _MockHooks(object):
+
+    def __init__(self):
+        self.called = 0
+  
+
+    def zap_started(self, zap, target):
+        self.called += 1
+        return zap,
+
+
+class TestZapHooks(unittest.TestCase):
+
+    @contextmanager
+    def assertNotRaises(self, exc_type=Exception):
+        try:
+            yield None
+        except exc_type:
+            raise self.failureException('{} raised'.format(exc_type.__name__))
+
+
+    def test_trigger_hook_arg_length(self):
+        """ Triggering hook should return same args if no hook to trigger """
+        self.assertEqual(zap_common.zap_hooks, None)
+        args = [1, 2, 3, 4]
+        return_args = zap_common.trigger_hook('print_rules_wrap', *args)
+        self.assertEqual(len(args), len(return_args))
+
+
+    def test_trigger_hook_mismatch_no_exception(self):
+        """ If the hook signature doesn't match the hook, no exception is raised """
+        zap_common.zap_hooks = _MockHooks()
+        with self.assertNotRaises():
+          zap_common.trigger_hook('zap_started')
+        self.assertEqual(zap_common.zap_hooks.called, 0)
+
+
+    def test_trigger_hook_ignore_exception(self):
+        """ Hooks that raise exceptions are ignored when triggered """
```

Comment:
This does not seem to be the best behaviour IMO, by continuing issues in the hooks could easily go unnoticed and the script would not be doing what the user expects/wants.